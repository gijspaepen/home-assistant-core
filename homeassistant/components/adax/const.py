"""Constants for the Adax integration."""

from typing import Final

from homeassistant.const import Platform

ACCOUNT_ID: Final = "account_id"
CLOUD = "Cloud"
CONNECTION_TYPE = "connection_type"
DOMAIN: Final = "adax"
LOCAL = "Local"
WIFI_SSID = "wifi_ssid"
WIFI_PSWD = "wifi_pswd"

PLATFORMS = [Platform.CLIMATE, Platform.SENSOR]
