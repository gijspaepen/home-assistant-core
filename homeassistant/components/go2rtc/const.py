"""Go2rtc constants."""

DOMAIN = "go2rtc"

CONF_DEBUG_UI = "debug_ui"
EXCLUSE_ERROR_MESSAGE = "Url and debug_ui cannot be set at the same time."
