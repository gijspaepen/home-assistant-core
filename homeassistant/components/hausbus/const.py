"""Constants for the Haus-Bus integration."""

from __future__ import annotations

import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "hausbus"
ATTR_ON_STATE = "on_state"
