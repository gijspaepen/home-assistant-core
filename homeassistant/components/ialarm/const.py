"""Constants for the iAlarm integration."""

from pyialarm import IAlarm

from homeassistant.components.alarm_control_panel import AlarmControlPanelEntityState

DATA_COORDINATOR = "ialarm"

DEFAULT_PORT = 18034

DOMAIN = "ialarm"

IALARM_TO_HASS = {
    IAlarm.ARMED_AWAY: AlarmControlPanelEntityState.ARMED_AWAY,
    IAlarm.ARMED_STAY: AlarmControlPanelEntityState.ARMED_HOME,
    IAlarm.DISARMED: AlarmControlPanelEntityState.DISARMED,
    IAlarm.TRIGGERED: AlarmControlPanelEntityState.TRIGGERED,
}
