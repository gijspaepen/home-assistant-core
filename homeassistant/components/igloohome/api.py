class Api:  # noqa: D100, D101
    def __init__(self) -> None:  # noqa: D107
        pass
