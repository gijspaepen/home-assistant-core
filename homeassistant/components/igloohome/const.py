"""Constants for the igloohome integration."""

DOMAIN = "igloohome"

OAUTH2_AUTHORIZE = "https://auth.igloohome.co/login"
OAUTH2_TOKEN = "https://auth.igloohome.co/oauth2/token"
