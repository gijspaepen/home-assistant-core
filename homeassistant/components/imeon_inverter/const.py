"""Constant for Imeon component."""

DOMAIN = "imeon_inverter"
HUBNAME = "imeon_inverter_hub"
TIMEOUT = 10  # seconds
