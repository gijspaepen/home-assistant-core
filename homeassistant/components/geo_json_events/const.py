"""Define constants for the GeoJSON events integration."""

from __future__ import annotations

from typing import Final

from homeassistant.const import Platform

DOMAIN: Final = "geo_json_events"

PLATFORMS: Final = [Platform.GEO_LOCATION]

ATTR_EXTERNAL_ID: Final = "external_id"
DEFAULT_RADIUS_IN_KM: Final = 20.0
DEFAULT_RADIUS_IN_M: Final = 20000.0
DEFAULT_UPDATE_INTERVAL: Final = 300
SOURCE: Final = "geo_json_events"
CONF_UPDATE_INTERVAL: Final = "update_interval"

SIGNAL_DELETE_ENTITY: Final = "geo_json_events_delete_{}"
SIGNAL_UPDATE_ENTITY: Final = "geo_json_events_update_{}"
