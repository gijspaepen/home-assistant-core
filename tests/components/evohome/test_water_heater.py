"""The tests for water_heater entities of evohome.

Not all evohome systems will have a DHW zone.
"""

from __future__ import annotations

from unittest.mock import patch

from freezegun.api import FrozenDateTimeFactory
import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.evohome.water_heater import EvoDHW
from homeassistant.core import HomeAssistant

from .const import TEST_INSTALLS_WITH_DHW


@pytest.mark.parametrize("install", TEST_INSTALLS_WITH_DHW)
async def test_set_operation_mode(
    hass: HomeAssistant,
    install: str,
    dhw: EvoDHW,
    freezer: FrozenDateTimeFactory,
    snapshot: SnapshotAssertion,
) -> None:
    """Test water_heater services of a evohome-compatible DHW zone."""

    freezer.move_to("2024-07-10T11:55:00Z")
    results = []

    # set_operation_mode(auto): FollowSchedule
    with patch("evohomeasync2.hotwater.HotWater._set_mode") as mock_fcn:
        await dhw.async_set_operation_mode("auto")

        assert mock_fcn.await_count == 1
        assert mock_fcn.await_args.args == (
            {
                "mode": "FollowSchedule",
                "state": None,
                "untilTime": None,
            },
        )
        assert mock_fcn.await_args.kwargs == {}

    # set_operation_mode(off): TemporaryOverride, advanced
    with patch("evohomeasync2.hotwater.HotWater._set_mode") as mock_fcn:
        await dhw.async_set_operation_mode("off")

        assert mock_fcn.await_count == 1
        assert install != "default" or mock_fcn.await_args.args == (
            {
                "mode": "TemporaryOverride",
                "state": "Off",
                "untilTime": "2024-07-10T12:00:00Z",  # varies by install
            },
        )
        assert mock_fcn.await_args.kwargs == {}

        results.append(mock_fcn.await_args.args)

    # set_operation_mode(on): TemporaryOverride, advanced
    with patch("evohomeasync2.hotwater.HotWater._set_mode") as mock_fcn:
        await dhw.async_set_operation_mode("on")

        assert mock_fcn.await_count == 1
        assert install != "default" or mock_fcn.await_args.args == (
            {
                "mode": "TemporaryOverride",
                "state": "On",
                "untilTime": "2024-07-10T12:00:00Z",  # varies by install
            },
        )
        assert mock_fcn.await_args.kwargs == {}

        results.append(mock_fcn.await_args.args)

    assert results == snapshot


@pytest.mark.parametrize("install", TEST_INSTALLS_WITH_DHW)
async def test_turn_away_mode_off(
    hass: HomeAssistant,
    install: str,
    dhw: EvoDHW,
) -> None:
    """Test water_heater services of a evohome-compatible DHW zone."""

    # turn_away_mode_off(): FollowSchedule
    with patch("evohomeasync2.hotwater.HotWater._set_mode") as mock_fcn:
        await dhw.async_turn_away_mode_off()

        assert mock_fcn.await_count == 1
        assert mock_fcn.await_args.args == (
            {
                "mode": "FollowSchedule",
                "state": None,
                "untilTime": None,
            },
        )
        assert mock_fcn.await_args.kwargs == {}


@pytest.mark.parametrize("install", TEST_INSTALLS_WITH_DHW)
async def test_turn_away_mode_on(
    hass: HomeAssistant,
    install: str,
    dhw: EvoDHW,
) -> None:
    """Test water_heater services of a evohome-compatible DHW zone."""

    # turn_away_mode_on(): PermanentOverride, Off
    with patch("evohomeasync2.hotwater.HotWater._set_mode") as mock_fcn:
        await dhw.async_turn_away_mode_on()

        assert mock_fcn.await_count == 1
        assert mock_fcn.await_args.args == (
            {
                "mode": "PermanentOverride",
                "state": "Off",
                "untilTime": None,
            },
        )
        assert mock_fcn.await_args.kwargs == {}


@pytest.mark.parametrize("install", TEST_INSTALLS_WITH_DHW)
async def test_turn_off(
    hass: HomeAssistant,
    install: str,
    dhw: EvoDHW,
) -> None:
    """Test water_heater services of a evohome-compatible DHW zone."""

    # turn_off(): PermanentOverride, Off
    with patch("evohomeasync2.hotwater.HotWater._set_mode") as mock_fcn:
        await dhw.async_turn_off()

        assert mock_fcn.await_count == 1
        assert mock_fcn.await_args.args == (
            {
                "mode": "PermanentOverride",
                "state": "Off",
                "untilTime": None,
            },
        )
        assert mock_fcn.await_args.kwargs == {}


@pytest.mark.parametrize("install", TEST_INSTALLS_WITH_DHW)
async def test_turn_on(
    hass: HomeAssistant,
    install: str,
    dhw: EvoDHW,
) -> None:
    """Test water_heater services of a evohome-compatible DHW zone."""

    # turn_on(): PermanentOverride, On
    with patch("evohomeasync2.hotwater.HotWater._set_mode") as mock_fcn:
        await dhw.async_turn_on()

        assert mock_fcn.await_count == 1
        assert mock_fcn.await_args.args == (
            {
                "mode": "PermanentOverride",
                "state": "On",
                "untilTime": None,
            },
        )
        assert mock_fcn.await_args.kwargs == {}
