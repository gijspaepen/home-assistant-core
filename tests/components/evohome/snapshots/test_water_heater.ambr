# serializer version: 1
# name: test_set_operation_mode[default]
  list([
    dict({
      'until': datetime.datetime(2024, 7, 10, 12, 0, tzinfo=datetime.timezone.utc),
    }),
    dict({
      'until': datetime.datetime(2024, 7, 10, 12, 0, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
